### DISCUSSION TITLE
How long should we support Python 3.7?

### DISCUSSION THREAD

**Discussion by falkoschindler** [zauberzeug/nicegui#982] [Polls]
Python 3.7 is currently the oldest version supported by NiceGUI. Due to its limited typing support, we need to take extra care here and there, which is a bit more work, but not too bad. The fact that other libraries stop supporting it might be a bigger problem.

On June 27, 2023, Python 3.7 will reach its [end-of-life](https://endoflife.date/python). This might be a good time to think about when we want to drop support for it. What do you think?

**Comment by pythonormen** [comment]
Anything that can free up time for you guys :)

**Comment by falkoschindler** [comment]
Thanks everyone for taking part in this little survey!

We come to the conclusion that the upcoming release of NiceGUI 1.3 is a good opportunity to discard Python 3.7. It simplifies things a little bit on our side, e.g. when working on new features like #1010. And besides that we don't want to encourage the community to stick with outdated and possibly vulnerable Python runtimes.
